File: java/Medium/_647_PalindromicSubstrings.java
package Medium;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.testng.annotations.Test;

public class _647_PalindromicSubstrings {
    public int countSubstrings(String s) {
        int res = 1;
        if(s== null || s.length() == 0) return 0;
        for(int i = 1; i<s.length(); i++){
            res += rec(s,i,i);
            res += rec(s,i-1,i);
        }

        return res;
    }

    private int rec(String s, int l, int r){
        int res = 0;
        while(l>=0 && r<s.length() && s.charAt(r)==s.charAt(l)){
            res++; l--; r++;
        }  
        return res;
    }

    @Test
    public void test(){
        assertEquals(6,countSubstrings("aaa"));
    }
}

File: Python3/_647_PalindromicSubstrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        if len(s) == 0:
            return 0
        count = 1    
        for i in range(1, len(s)):
            count += self.rec(s,i,i)
            count += self.rec(s,i-1,i)
        return count
    
    def rec(self,s,l,r) -> int:
        res = 0
        while l>=0 and r < len(s) and s[l] == s[r]:
            res += 1
            l -= 1
            r += 1
        return res


test = Solution()
print(test.countSubstrings("aaa"))

    